### PR TITLE
feat(discv2): query avatars 

### PR DESCRIPTION
**Pull request by doralchan**
**Before:**
Screen Shot 2020-03-30 at 9 56 42 AM

**After:** 
Screen Shot 2020-03-31 at 12 52 15 PM

### CODE REVIEW

File: src/sentry/static/sentry/app/views/eventsV2/querycard.tsx [hunk 1]
```diff
@@ -69,6 +59,20 @@ class QueryCard extends React.PureComponent<Props> {
   }
 }
 
+const QueryCardContent = styled('div')`
+  flex-grow: 1;
+  overflow: hidden;
+  margin-right: ${space(1)};
+`;
+
+const Avatar = styled('div')`
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/querycard.tsx]: Is this going to contain some user related content soon?

Comment (doralchan) [reply src/sentry/static/sentry/app/views/eventsV2/querycard.tsx]: Yep! Just trying to work with Will to get it 👍

File: src/sentry/api/serializers/models/discoversavedquery.py [hunk 2]
```diff
@@ -32,7 +33,9 @@ def serialize(self, obj, attrs, user, **kwargs):
             "version": obj.version or obj.query.get("version", 1),
             "dateCreated": obj.date_created,
             "dateUpdated": obj.date_updated,
-            "createdBy": six.text_type(obj.created_by_id) if obj.created_by_id else None,
+            "createdBy": serialize(obj.created_by, serializer=UserSerializer())
+            if obj.created_by
+            else None,
```
Comment (markstory) [inline src/sentry/api/serializers/models/discoversavedquery.py]: We should check that this doesn't break discover1. While its not long for the world we shouldn't break it before we remove it.

File: src/sentry/static/sentry/app/views/eventsV2/querycard.tsx [hunk 3]
```diff
@@ -39,23 +40,24 @@ class QueryCard extends React.PureComponent<Props> {
...
-            </CardHeading>
-            <StyledQueryDetail>{queryDetail}</StyledQueryDetail>
+            <QueryCardContent>
+              <QueryTitle>{title}</QueryTitle>
+              <QueryDetail>{queryDetail}</QueryDetail>
+            </QueryCardContent>
+            {starred ? (
+              <StyledUserAvatar user={createdBy} />
+            ) : (
+              <Avatar>
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/querycard.tsx]: There is another `SystemAvatar` in app/components/activity/item/avatar.tsx. Would you want to consolidate that with this?

Comment (doralchan) [reply src/sentry/static/sentry/app/views/eventsV2/querycard.tsx]: What's the difference between SystemAvatar and UserAvatar in this sense?

File: src/sentry/api/serializers/models/discoversavedquery.py (hunk 2 above)
Comment (wmak) [reply src/sentry/api/serializers/models/discoversavedquery.py]: :+1: Clicked around v1 and it looks good. Grepping through the static code createdBy is included in a few spots but not used AFAICT.

File: src/sentry/api/serializers/models/discoversavedquery.py [hunk 4]
```diff
@@ -32,7 +33,9 @@ def serialize(self, obj, attrs, user, **kwargs):
             "version": obj.version or obj.query.get("version", 1),
             "dateCreated": obj.date_created,
             "dateUpdated": obj.date_updated,
-            "createdBy": six.text_type(obj.created_by_id) if obj.created_by_id else None,
+            "createdBy": serialize(obj.created_by, serializer=UserSerializer())
```
Comment (markstory) [inline src/sentry/api/serializers/models/discoversavedquery.py]: @wmak Do we eager load this relationship or are we adding an N query?

Comment (wmak) [reply src/sentry/api/serializers/models/discoversavedquery.py:36]: In the case of a post to the saved query endpoint, this is a `request.user`'s `SimpleLazyObject` so I think we add a query in that case.
For a get I think this was already loaded.

Comment (markstory) [reply src/sentry/api/serializers/models/discoversavedquery.py:36]: The only prefetch_related I see on the GET is for projects.

Comment (wmak) [reply src/sentry/api/serializers/models/discoversavedquery.py:36]: Ah you're right, i'll add a `select_related` to the query :+1:

**Review by markstory — APPROVED** [review]

**Pull request merged**